src/Components/ProvisioningWizard/index.js

```diff
@@ -20,9 +21,12 @@ const ProvisioningWizard = ({ isOpen, onClose, image, ...props }) => {
   const [stepValidation, setStepValidation] = React.useState(DEFAULT_STEP_VALIDATION);
   const [isConfirming, setConfirming] = React.useState(false);
   const [successfulLaunch, setLaunchSuccess] = React.useState();
-
   const { isLoading, error: sourcesError, sources: availableSources } = sourcesForImage(image, { refetch: stepIdReached <= 2 });
 
+  React.useEffect(() => {
+    axios.defaults.headers.common['Provisioning-Wizard-Id'] = uuidv4();
```

Comment:
what about adding a cleanup when the wizard unmounts? just to be safe that we don't change axios globally. 
```js
useEffect(() => {
  performSideEffect();
  return cleanUpFunction;
}, []);
```
We need also to be carefull due to react 18, which `useEffect` triggers twice (so we might end up with two distinct ids on the same launch)

I'm not sure what is the goal here, something like session id?

Reply:
The goal is just to have a way to correlate all the requests from a single wizard.
Meaning to have a way in the backend to say what requests came from single wizard session (so yeah something like a session, but just a Wizard session).

Reply:
@amirfefer Do you have a different solution than using the useEffect?

Reply:
there are some solutions to bypass this issue, i.e create a [custom hook](https://dev.to/ag-grid/react-18-avoiding-use-effect-getting-called-twice-4i9e) which trigger useEffect only once.